Add "Lotus install" section. (#235)
filecoin-project/filecoin-docs: docs/install/lotus/troubleshooting.md

---
title: Troubleshooting Lotus installation

> description: If you're having issues install Lotus on your system, try running through some of these troubleshooting steps.

Reviewer: ```suggestion
description: If you're having issues installing Lotus on your system, try running through some of these troubleshooting steps.
```

---

# Troubleshooting

> If you're having trouble installing Lotus, try running through some of these troubleshooting steps. If you're still having trouble, feel free to reach out to the team on the `#fil-help` channel in the Filecoin Slack.

Reviewer: ```suggestion
If you're having trouble installing Lotus, try running through some of these troubleshooting steps. If you're still having trouble, feel free to reach out on the `#fil-help` channel in the Filecoin Slack.
```

encourage community support too

## Update your system

> This sounds simple, but a common reason users aren't able to run Filecoin is because their system isn't up to date.

Reviewer: Does this mean operating system? hardware system?

Author: Operating system.

## Check the dependencies

> Lotus has several dependencies that need to be installed and working _before_ the software can run. Run the command below for your operating system to make sure everything is installed and up-to-date.

Reviewer: ```suggestion
Lotus has several dependencies that must be installed _before_ the software can run. Run the command below for your operating system to make sure everything is installed and up-to-date.
```

## Common errors

Here are some common errors users have encountered, and how to fix them.

### Go: command not found

You may encounter the following error:

```bash
bash: go: command not found
make: *** [Makefile:68: lotus] Error 127
```

This error means that you do not have Go installed properly. Run `go version` to see which version of Go you have installed. If you don't get a response, Go has not been installed correctly. If you _think_ you've already downloaded and installed Go, check that the Go binary has been added to your system PATH. You may need to reboot your machine after adding the Go binary to your system PATH. See the [Go documentation for more details](https://golang.org/doc/install#install).

### panic: qtls.ClientSessionState not compatible with tls.ClientSessionState

You may encounter the following error:

```bash
panic: qtls.ClientSessionState not compatible with tls.ClientSessionState
goroutine 1 [running]:
github.com/lucas-clemente/quic-go/internal/handshake.init.2()
	/home/infra/go/pkg/mod/github.com/lucas-clemente/quic-go@v0.17.3/internal/handshake/unsafe.go:26 +0x205
```

> The current Lotus client `0.4.2` has issues with Go version `1.15`. If you have Go `1.15` installed, roll back to version `1.14.6`.

Reviewer: ```suggestion
The current Lotus client `0.4.2` is not compatible with Go version `1.15`. If you have Go `1.15` installed, roll back to version `1.14.6`.
```
